packages/models-library/src/models_library/services_types.py

```diff
@@ -1,18 +1,28 @@
-from typing import Annotated, Any, TypeAlias
+from typing import TYPE_CHECKING, Annotated, Any, Self, TypeAlias
 from uuid import uuid4
 
 import arrow
-from pydantic import GetCoreSchemaHandler, StringConstraints, ValidationInfo
+from pydantic import (
+    GetCoreSchemaHandler,
+    PositiveInt,
+    StringConstraints,
+    ValidationInfo,
+)
 from pydantic_core import CoreSchema, core_schema
 
 from .basic_regex import PROPERTY_KEY_RE, SIMPLE_VERSION_RE
+from .projects_nodes_io import NodeID
 from .services_regex import (
     COMPUTATIONAL_SERVICE_KEY_RE,
     DYNAMIC_SERVICE_KEY_RE,
     FILENAME_RE,
     SERVICE_ENCODED_KEY_RE,
     SERVICE_KEY_RE,
 )
+from .users import UserID
+
+if TYPE_CHECKING:
+    from .projects import ProjectID
```

Comment:
I wonder why this this necessary? Is it because of cyclic dependencies?

TIP: For things like `ProjectID` we can always move these to `common_library`.  I already did that with user_enums and groups_enums (check how it is imported in the Groups and Users models)